File: apps/api/src/main/kotlin/com/wewiins/saas_api/utils/Sanitize.kt
package com.wewiins.saas_api.utils

import java.text.Normalizer

class Sanitize {
    companion object {

        fun text(value: String): String {
            return Normalizer.normalize(value, Normalizer.Form.NFD)
                .replace(Regex("\\p{InCombiningDiacriticalMarks}+"), "")
                .lowercase()
                .replace(Regex("[@.]"), "_")
                .replace(Regex("\\s+"), "_")
                .replace(Regex("[^a-z0-9_-]"), "")
                .replace(Regex("_+"), "_")
                .trim('_')
        }
    }
}

File: apps/client/src/app/utils/sanitize.ts
export function sanitize(value: string): string {
    return value
        .toLowerCase()
        .replace(/[@.&]/g, '_')
        .replace(/\s+/g, '_')
        .replace(/_+/g, '_')
        .replace(/[^a-z0-9_-]/g, '');
}